JAICore/jaicore-ml/src/jaicore/ml/dyadranking/algorithm/PLNetDyadRanker.java

```diff
@@ -1,43 +1,381 @@
+			int numOut = hiddenNodes.get(i + 1);
+			configBuilder.layer(i + 1, new DenseLayer.Builder()
+					.nIn(numIn)
+					.nOut(numOut)
+					.weightInit(WeightInit.XAVIER)
+					.activation(Activation.fromString(activation))
+					.build());
+		}
+
+		// Build output layer. Since we are using an external error for training,
+		// this is a regular layer instead of an OutputLayer
+		configBuilder.layer(hiddenNodes.size(), new DenseLayer.Builder()
+				.nIn(hiddenNodes.get(hiddenNodes.size() - 1))
+				.nOut(1)
+				.weightInit(WeightInit.XAVIER)
+				.activation(Activation.IDENTITY)
+				.build());
+
+		MultiLayerConfiguration multiLayerConfig = configBuilder.build();
+		return new MultiLayerNetwork(multiLayerConfig);
+	}
+	
+	/**
+	 * Converts a dyad to a {@link INDArray} row vector consisting of a concatenation of the instance and alternative features.
+	 * 
+	 * @param dyad	The dyad to convert.
+	 * @return		The dyad in {@link INDArray} row vector form.
+	 */
+	private INDArray dyadToVector(Dyad dyad) {
+		INDArray instanceOfDyad = Nd4j.create(dyad.getInstance().asArray());
+		INDArray alternativeOfDyad = Nd4j.create(dyad.getAlternative().asArray());
+		INDArray dyadVector = Nd4j.hstack(instanceOfDyad, alternativeOfDyad);
+		return dyadVector;
+	}
+	
+	/**
+	 * Converts a dyad ranking to a {@link INDArray} matrix where each row corresponds to a dyad.
+	 * @param drInstance	The dyad ranking to convert to a matrix.
+	 * @return				The dyad ranking in {@link INDArray} matrix form.
+	 */
+	private INDArray dyadRankingToMatrix(IDyadRankingInstance drInstance) {
+		List<INDArray> dyadList = new ArrayList<INDArray>(drInstance.length());
+		for (Dyad dyad : drInstance) {
+			INDArray dyadVector = dyadToVector(dyad);
+			dyadList.add(dyadVector);
+		}
+		INDArray dyadMatrix;
+		dyadMatrix = Nd4j.vstack(dyadList);
+		return dyadMatrix;
+	}
+
+	/**
+	 * Creates a simple feed-forward {@link MultiLayerNetwork} using the json
+	 * representation of a {@link MultiLayerConfiguration} in the file .
+	 * 
+	 * @param configFile {@link File} containing the json representation of the
+	 *                   {@link MultiLayerConfiguration}
+	 */
+	public void createNetworkFromDl4jConfigFile(File configFile) {
+		String json = "";
+		try {
+			json = FileUtil.readFileAsString(configFile);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		MultiLayerConfiguration config = MultiLayerConfiguration.fromJson(json);
+		MultiLayerNetwork network = new MultiLayerNetwork(config);
+		this.plNet = network;
+	}
+
+	/**
+	 * Save a trained model at a given file path. Note that the produced file is a
+	 * zip file and a ".zip" ending is added.
+	 * 
+	 * @param filePath The file path to save to.
+	 * @throws IOException
+	 */
+	public void saveModelToFile(String filePath) throws IOException {
+		if (plNet == null) {
+			throw new IllegalStateException("Cannot save untrained model.");
+		}
+		File locationToSave = new File("filePath" + ".zip");
```

Comment:
This does not actually zip the file,[ ModelSerializer.writeModel just uses a BufferedOutputStream](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ModelSerializer.java ). To zip a file use a ZipOutPutStream ([Tutorial](https://www.baeldung.com/java-compress-and-uncompress)), or just change the file extension to not be zip.

Reply:
Seems to work without actually using the zip stream, my bad